### DISCUSSION TITLE
Beginner question to get started

### DISCUSSION THREAD

**Discussion by WemosAlaPlaya** [curl/curl#13021] [Q&A] [answered]
I'm new with CURL, but working with computers since 1982, so I think I know basic things. Trying several examples, but always get 'syntax error' referring to 'https'. So what is wrong here;
curl -X GET --header 'Accept: application/xml' --header 'app_id: XXX' --header 'app_key: XXX' 'https://api.finavia.fi/flights/public/v0/flights/all'

**Comment by dfandrich** [comment, marked as answer]
For one, you don't need `-X GET`; just drop it. Otherwise, the command should work, and runs for me (giving an authentication error, of course). You don't say what platform or version of curl you're using. Windows, for example, doesn't like single quote marks much. Some versions also use a fake curl that doesn't support real curl options.

**Comment by WemosAlaPlaya** [comment]
> For one, you don't need `-X GET`; just drop it. Otherwise, the command should work, and runs for me (giving an authentication error, of course). You don't say what platform or version of curl you're using. Windows, for example, doesn't like single quote marks much. Some versions also use a fake curl that doesn't support real curl options.

Windows is my platform.
So, all "help" I tried to find via google, does not help me. Also, people who knows much better about curl/api's, have not get this to work. Like I said, I'm beginner with this, so I do  not know how to make authentication via HTML/python or what ever it takes. Everything works fine when using "finavia", developer portal, and I can do the authentication via that, but there is no instructions how to do that via code.

**Reply by dfandrich** [reply]
You haven't given many details about the failure, so we have to try to fill in the blanks ourselves here. Adding the output of -v would help us to see where this "syntax error" is coming from, but if it's coming from the remote site (which is likely since curl won't report that error for this kind of command) then you'll have to refer to the site's documentation to find out why it doesn't like the request you're sending.

**Comment by WemosAlaPlaya** [comment]
> You haven't given many details about the failure, so we have to try to fill in the blanks ourselves here. Adding the output of -v would help us to see where this "syntax error" is coming from, but if it's coming from the remote site (which is likely since curl won't report that error for this kind of command) then you'll have to refer to the site's documentation to find out why it doesn't like the request you're sending.

Information I gave is all I have. And like I said, this is totally new for me, so I do not have a clue what you mean by this; "For one, you don't need -X GET; just drop it"
Of course I have to put GET command there.

Both, HTML and .py files are on my windows laptop (now also on my website, but does not give any response,  fail nor pass). Using command line (python), that I have understood should return the information, but I just get that 'syntax error', referring to https.
Thank you for your patience this far. I am not going to implement this application, only want to see if it works, when a guy I already paid, has not get this one to work. It sounded that is simple task for him.

**Reply by dfandrich** [reply]
I'm trying to give you advice to help you solve this, which you're free to take or not. But this is going to take debugging on your part because you haven't shown us how to reproduce the problem.

**Reply by WemosAlaPlaya** [reply]
I have understood that if I open python via windows command line and put that curl command as above there, it should be give me the answer/info that I want. Thats all. Nothing to debug. Just a one line.
Or, just right to the windows CMD, not python? This does not work either.
 I keep googling. Thanks this far.

**Reply by bagder** [reply]
> Of course I have to put GET command there.

[No you don't](https://daniel.haxx.se/blog/2015/09/11/unnecessary-use-of-curl-x/)

**Comment by dfandrich** [comment]
That explains the SyntaxError you're seeing: that's not a Python command, that's a shell command. On
Windows, open a shell (with Win-R cmd) and type it in there. Although, you'll
probably need to replace the single quotes with double quotes on Windows.

**Comment by WemosAlaPlaya** [comment]
Ah, now got it; --ssl-no-revoke. This was frustrating me so much and feeling myself stupid, but now it works.
